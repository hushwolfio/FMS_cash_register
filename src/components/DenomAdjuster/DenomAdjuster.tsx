import type { FC } from "react";
import { NumberInput } from "@mantine/core";
import type { Denominations } from "../../types";

interface ButtonProps {
	denom: keyof Denominations;
	value: number;
	setDenomAmount: (denomination: keyof Denominations, count: number) => void;
}

export const DenomAdjuster: FC<ButtonProps> = ({ denom, value, setDenomAmount }) => {
	const onInputChange = (value: string | number): void => {
		setDenomAmount(denom, Number(value));
	}

	return (
		<NumberInput
			defaultValue={0}
			value={value}
			label={`Add/Remove $${denom}`}
			onChange={onInputChange}
			allowDecimal={false}
			allowNegative={false}
			min={0}
			style={{
				width: 120
			}} />
	);
};
