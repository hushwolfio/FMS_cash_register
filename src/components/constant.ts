import type { Denominations } from "../types";

export const DENOMINATIONS_KEYS = ['20', '10', '5', '2', '1'];

export const INITIAL_DENOMINATIONS: Denominations = { '20': 0, '10': 0, '5': 0, '2': 0, '1': 0 };
