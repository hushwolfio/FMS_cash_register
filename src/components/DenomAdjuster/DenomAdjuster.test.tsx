import '../../matchMedia';
import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { DenomAdjuster } from './DenomAdjuster';
import { MantineProvider } from '@mantine/core';
import { DENOMINATIONS_KEYS } from '../constant';

describe('DenomAdjuster', () => {
    it('should render', () => {
        const denom = DENOMINATIONS_KEYS[0];
        const value = 1;
        const setDenomAmount = vi.fn();
        const { container } = render(<MantineProvider><DenomAdjuster denom={denom} value={value} setDenomAmount={setDenomAmount} /></MantineProvider>);
        expect(container).toMatchSnapshot();
    });
});