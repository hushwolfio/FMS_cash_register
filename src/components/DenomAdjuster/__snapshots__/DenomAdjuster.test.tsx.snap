// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DenomAdjuster > should render 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_e2f5cd4e m_46b77525 mantine-InputWrapper-root mantine-NumberInput-root"
    style="width: 120px;"
  >
    <label
      class="m_8fdc1311 mantine-InputWrapper-label mantine-NumberInput-label"
      for="mantine-m7l9yhvfu"
      id="mantine-m7l9yhvfu-label"
    >
      Add/Remove $20
    </label>
    <div
      class="m_6c018570 mantine-Input-wrapper mantine-NumberInput-wrapper"
      data-variant="default"
      data-with-right-section="true"
      style="--input-right-section-width: var(--ni-right-section-width-sm);"
    >
      <input
        aria-invalid="false"
        class="m_8fb7ebe7 mantine-Input-input mantine-NumberInput-input"
        data-variant="default"
        id="mantine-m7l9yhvfu"
        inputmode="numeric"
        type="text"
        value="1"
      />
      <div
        class="m_82577fc2 mantine-Input-section mantine-NumberInput-section"
        data-position="right"
      >
        <div
          class="m_95e17d22 mantine-NumberInput-controls"
        >
          <button
            aria-hidden="true"
            class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
            data-direction="up"
            tabindex="-1"
            type="button"
          >
            <svg
              fill="none"
              style="transform: rotate(180deg);"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            aria-hidden="true"
            class="mantine-focus-auto m_80b4b171 mantine-NumberInput-control m_87cf2631 mantine-UnstyledButton-root"
            data-direction="down"
            tabindex="-1"
            type="button"
          >
            <svg
              fill="none"
              viewBox="0 0 15 15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
